part_1.py

```diff
@@ -16,17 +16,131 @@ def main():
     fs = hadoop.fs.FileSystem
     conf = hadoop.conf.Configuration()
     path = hadoop.fs.Path('/user/hm74/NYCOpenData')
-
     for nyc_open_datafile in fs.get(conf).listStatus(path)[0:2]:
         # pretty hacky preprocessing but it will work for now
         # could maybe use pathlib library or get it with hdfs
         processed_path = str(nyc_open_datafile.getPath()).replace("hdfs://dumbo", "")
-        print(processed_path)
         df_nod = spark.read.option("header", "true").option("delimiter", "\t").csv(processed_path)
-        bp = BasicProfiling(df_nod)
-        type_dict = bp.find_all_column_types()
+        bp = BasicProfiling(processed_path, df_nod)
+        type_dict = bp.process()
         print(type_dict)
 
 
+# We should put this in it's on package, but submitting with packages is kind of annoying so
+# I moved it out for now look at --py-files
+#https://spark.apache.org/docs/latest/submitting-applications.html
+class BasicProfiling:
+    """
+    Class for data profiling basic schema and statistics on a dataframe
+    """
+    def __init__(self, dataset_name, df_nod):
+        self.dataset_name = dataset_name
+        self.df_nod = df_nod
+        self.table_dict = dict()
+        self.columns = self.df_nod.columns
+        self.spec_types = ['INT', 'REAL', 'DATE', 'TEXT']
+
+    def __set_up_dictionary(self):
+        self.table_dict['dataset_name'] = self.dataset_name
+        self.table_dict['columns'] = self.columns
+        self.table_dict['column_specification'] = []
```

Comment:
@mrpoor I think you misunderstand the output format here. We don't need the "column_specification" as a key. My understand is "columns" is a list, and "column_specification" is an element of the list. I will merge this request first. If you think I'm right I will modify this later.

Reply:
@JunqiLoveCoding 
I mean think about it you have json, you can't have key, value json and then just a list.

The following is what you think it is, it is not valid json:
```
{
	"name": "greg",
	"another": "value",
	["somevalue", "someothervalue"]
}
```

Reply:
But there should not be a "value" in "another", there should just be a list.